### PR TITLE
Papercuts

### PR DESCRIPTION
**Pull request by symroe**
This is a pick and mix of commits that fix small things without much in the way of functional changes.

A load of them are about fixing up warnings about deprecations. Some are fixing #643.

Best to take each commit on it's own

### CODE REVIEW

File: ynr/apps/candidates/views/candidacies.py [hunk 1]
```diff
@@ -81,17 +81,6 @@ def form_valid(self, form):
             person.save()
         return get_redirect_to_post(self.election, post)
 
-    def get_context_data(self, **kwargs):
```
[the remark this answers is not in the document]
Comment (symroe) [reply ynr/apps/candidates/views/candidacies.py]: 👍

**Comment by symroe** [comment]
> I did this on WhereDIV and it was not a big job: DemocracyClub/UK-Polling-Stations@6112cc8 (but there were only 2 of them and they were quite simple). If we're not going to do this here, lets raise it as an issue for future.

I've done it here as a fixup, but TBH I think this is worse than just using the middleware mixin. I don't like that Django is making me write duplicate boilerplate code. I get we don't want to use `django.utils.deprecation`, but I reckon Django should provide a mixin for this (hence why I guess they are using `django.utils.deprecation` themselves)

I thought about writing a mixin to do this in this codebase, but then I'd just be duplicating the Django mixin.

 `¯\_(ツ)_/¯`

**Comment by symroe** [comment]
CI failing because Black updated its version (and therefore the Travis version) between opening this PR and pushing updates.

I expect we should pin the Black version at some point, but this has actually caught some useful formatting errors from the previous run, so I might keep it unpinned for a while until it's more stable.

**Comment by chris48s** [comment]
> I expect we should pin the Black version at some point, but this has actually caught some useful formatting errors from the previous run, so I might keep it unpinned for a while until it's more stable.

If we can get production running on python 3.6 then we don't need to run the tests on python 3.5 anymore. Then we can just pin it in the requirements, let dependabot update it (like we do with everything else) and stop manually installing it in CI. That's probably the situation we want to arrive at.

**Pull request merged**